### PR TITLE
deps: define Beam version as a Maven property in pom

### PR DESCRIPTION
**Pull request by dpcollins-google**

### CODE REVIEW

File: samples/snippets/pom.xml [hunk 1]
```diff
@@ -61,7 +62,7 @@
     <dependency>
       <groupId>org.apache.beam</groupId>
       <artifactId>beam-model-pipeline</artifactId>
-      <version>2.29.0</version>
+      <version>${beam.version}</version>
     </dependency>
     <!-- [END pubsublite_java_dependencies] -->
```
Comment (anguillanneuf) [inline samples/snippets/pom.xml:67]: Need to move the region tags to include only `google-cloud-pubsublite` and `google-cloud-pubsublit`, these show up on https://cloud.google.com/pubsub/lite/docs/reference/libraries#client-libraries-install-java, and is for running Lite samples only.

Deps for cross-product samples should not be included by this region tag.

**Review by anguillanneuf — CHANGES_REQUESTED** [review]
Samples tests failed.

```
java.lang.NoSuchMethodError: 'org.apache.beam.vendor.grpc.v1p26p0.com.google.protobuf.Descriptors$EnumValueDescriptor org.apache.beam.model.pipeline.v1.RunnerApi$BeamConstants$Constants.getValueDescriptor()'
	at pubsublite.snippets@1.0.23/pubsublite.beam.BeamReadWriteIT.testQuickstart(BeamReadWriteIT.java:147)
```

File: samples/snippets/pom.xml (hunk 1 above)
Comment (dpcollins-google) [reply samples/snippets/pom.xml]: Done

**Comment by dpcollins-google** [comment]
> Samples tests failed.
> 
> ```
> java.lang.NoSuchMethodError: 'org.apache.beam.vendor.grpc.v1p26p0.com.google.protobuf.Descriptors$EnumValueDescriptor org.apache.beam.model.pipeline.v1.RunnerApi$BeamConstants$Constants.getValueDescriptor()'
> 	at pubsublite.snippets@1.0.23/pubsublite.beam.BeamReadWriteIT.testQuickstart(BeamReadWriteIT.java:147)
> ```

This seems like it only happens when 2.30 is specified for beam when using the I/O built against 2.29. I'll look into how we can resolve this in the future, although I'd note that you can use the I/O built against 2.29 with as far back as 2.26 by my testing

**Comment by anguillanneuf** [comment]
This error is preventing other PRs from passing all tests:

```
Error:  Failures: 
Error:    SubscriptionPartitionProcessorImplTest.subscriberFailureFails:142 expected com.google.api.gax.rpc.ApiException to be thrown, but nothing was thrown
```

**Review by anguillanneuf — APPROVED** [review]

**Pull request merged**